functions/visualizations.py

```diff
@@ -79,9 +79,88 @@ def _get_transforms() -> dict:
     return {
         "clustergrammer": _ClustergrammerTransform(),
         "ihc_combined_plot": _ihc_combined_transform,
+        "additional_metadata": _add_antibody_metadata,
     }
 
 
+def _add_antibody_metadata(
+    file_record: DownloadableFiles, metadata_df: pd.DataFrame
+) -> Optional[dict]:
+    """
+    Pseudo transformation to add antibody data to the DownloadableFiles.additional_metadata JSON
+    Only for upload_type in [cytof, elisa, ihc, micsss, and mif]
+    """
+    upload_type = file_record.upload_type.lower()
+    if upload_type not in ["cytof", "elisa", "ihc", "micsss", "mif"]:
+        return None
+
+    file_md = file_record.additional_metadata
+
+    ct_md = TrialMetadata.find_by_trial_id(file_record.trial_id).metadata_json
+    assay_md = ct_md.get("assays", {}).get(file_record.upload_type.lower(), {})
+
+    # differences in data model
+    if upload_type == "cytof":
+        key = "cytof_antibodies"
+    elif upload_type in ["elisa", "mif"]:
+        key = "antibodies"
+    elif upload_type in ["ihc", "micsss"]:
+        key = "antibody"
+    else:  # shouldn't get here
+        return None
+
+    antibody_md = assay_md.get(key)
+
+    if not antibody_md:
+        return None
+
+    if upload_type in ["cytof", "elisa"]:
+        antibodies = []
+        for ab in antibody_md:
+            if ab["usage"] != "Ignored":
+                entry = f"{ab['stain_type'].lower().split()[0]} {ab['isotope']}-{ab['antibody']}"
```

Comment:
`some_string.split()[0] == some_string` for any string `some_string` except the empty string, which `KeyError`s. Did you mean to supply an argument to `split`?

Reply:
`split()` is equivalent to `split(' ')`, so this phrasing returns only the first word in the string
the valid entries here (from schema: [cytof](https://github.com/CIMAC-CIDC/cidc-schemas/blob/master/cidc_schemas/schemas/assays/components/cytof/cytof_antibody.json#L27), [elisa](https://github.com/CIMAC-CIDC/cidc-schemas/blob/master/cidc_schemas/schemas/assays/components/elisa/elisa_antibody.json#L27)) are "Surface Stain" and "Intercellular" so after conversion "Surface" and "Intercellular"

Reply:
Oh, whoops - guess my above statement was totally false. Sounds good!